packages/babel-plugin-transform-react-constant-elements/src/index.js

```diff
@@ -53,6 +55,16 @@ export default function ({ types: t }) {
         if (path.node._hoisted) return;
 
         const state = { isImmutable: true };
+
+        // This transform takes the option `allowMutablePropsOnTags`, which is an array
+        // of JSX tags to allow mutable props (such as objects, functions) on. Use sparingly
+        // and only on tags you know will never modify their own props.
+        if (this.opts && Array.isArray(this.opts.allowMutablePropsOnTags)) {
```

Comment:
I think you can assume `this.opts` is always an object. 

My preference is to be strict on arguments, so I'd prefer

```
if (this.opts.allowMutablePropsOnTags != null && !Array.isArray(this.opts.allowMutablePropsOnTags)) {
  throw new Error(".allowMutablePropsOnTags must be an array, null, or undefined.");
}
```